### PR TITLE
Enable mp3 output on esp32s3

### PR DESCRIPTION
**Pull request by jepler**
With these changes I can successfully play a 48kBit/s, 22.5kHz, mono test file on the Adafruit Feather ESP32-S3 TFT using a max98357 breakout, which takes less than 10% of available CPU times. I have not tried higher bit rates, stereo playback etc., so I don't know the performance limits.

There are occasional audio glitches, which mostly correlate with the feather's charging LED. I think this is a HW problem but it's difficult to be sure.

This also fixes the problem that *all* I2S output was in a non-standard format since updating esp-idf (to 5.0? 5.1?). Now, the "philips standard" format is used again.

This needs https://github.com/adafruit/Adafruit_MP3/pull/21

I made sure that my test program **does not generate garbage while playing**. The maximum I2S DMA buffer size is only about 1000 samples, or only about 20ms @ 48kHz. blocking background tasks *will* make the audio glitch, and garbage collection blocks background tasks. This is a larger problem on microcontrollers with PSRAM, like the feather s3 series, because the collection time is proportional to the size of the GC heap.

This also doesn't work on a debug build, the different optimization flags or additional error checking seemed to just add too much overhead to play back without glitching.

### CODE REVIEW

File: ports/espressif/mpconfigport.mk [hunk 1]
```diff
@@ -136,6 +135,7 @@ CIRCUITPY_BLEIO ?= 0
 else
 CIRCUITPY_BLEIO ?= 1
 endif
+CIRCUITPY_AUDIOMP3 ?= 1
```
Comment (tannewt) [inline ports/espressif/mpconfigport.mk:138]: Why S3 only? I'd probably only enable it on boards with more than 4mb flash. There others are tight on space and this is a new feature.

**Review by tannewt — CHANGES_REQUESTED** [review]
One question about what boards to enable on. Good otherwise.

File: ports/espressif/mpconfigport.mk (hunk 1 above)
Comment (jepler) [reply ports/espressif/mpconfigport.mk:138]: I only tested on S3. Do you want me to test on other boards, or just enable widely?

Comment (tannewt) [reply ports/espressif/mpconfigport.mk:138]: I think you should try it everywhere. We do have size issues though.

Comment (jepler) [reply ports/espressif/mpconfigport.mk]: OK, the current build should enable it on anything with 4MB+

File: ports/espressif/mpconfigport.mk [hunk 2]
```diff
@@ -178,3 +182,7 @@ USB_NUM_IN_ENDPOINTS = 5
 
 # Usually lots of flash space available
 CIRCUITPY_MESSAGE_COMPRESSION_LEVEL ?= 1
+
+# Don't enable mp3 by default but DO allocate it to on-chip RAM if enabled
```
Comment (tannewt) [inline ports/espressif/mpconfigport.mk:186]:
```suggestion
```

**Review by tannewt — CHANGES_REQUESTED** [review]
One obsolete comment. Good otherwise. Thanks!

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**